### ISSUE TITLE
docs website connection not secure

### ISSUE THREAD

**Issue by swpease** [getpelican/pelican#2502]
I don't recall if I was just using Chrome before, which connects in spite of the insecure connection, but using Firefox:

>The owner of docs.getpelican.com has configured their website improperly. To protect your information from being stolen, Firefox has not connected to this website.

>This site uses HTTP Strict Transport Security (HSTS) to specify that Firefox may only connect to it securely. As a result, it is not possible to add an exception for this certificate.

**Comment by mosra** [comment]
On Chrome I was always landing on HTTP, e.g. http://docs.getpelican.com/en/stable/settings.html, tried to change that to HTTPS once but it yelled something awful at me, so I never tried again :sweat_smile:

**Comment by justinmayer** [comment]
This is normal and expected. All links pointing to docs.getpelican.com should already be prefixed with http:// since ReadTheDocs does not support HTTPS on custom domains.

[That is, until a few months ago, apparently.](http://blog.readthedocs.com/https-for-custom-domains/) While HTTPS support for docs.getpelican.com should have been automatic, it seems there might have been a glitch in that automatic transition. I just now edited and saved the domain in the RTD console, making no changes, which seems to have triggered the TLS certificate generation.

In short, accessing https://docs.getpelican.com/ should no longer result in certificate-related warnings. 🎉

**Comment by mosra** [comment]
@justinmayer wow, that was quick! :+1: 

The only thing missing is a redirect from http to https. Not sure how readthedocs handles that, tho (personally I was doing it with a `.htaccess` rule on my Apache installation).

**Comment by justinmayer** [comment]
I don't know if I'd say "missing" — it would be nice, but I don't think it's important. I've checked the `Always use HTTPS for this domain` box, so presumably RTD will manage the re-direction at some point.

**Comment by mosra** [comment]
Oh, okay, then let's hope it automagically fixes itself in a while :)

IIRC google is (slightly) penalizing results that are not HTTPS (and browsers are being increasingly picky about non-encrypted connections) so if you don't force-redirect to HTTPS, it won't have a chance to know you have HTTPS now, because HTTP still works and all links right now lead to HTTP. This is the info I gathered when I was doing the HTTPS switch for my own sites few months ago. I also don't think it's very important, but if it can improve page rank and hide the annoying "Unencrypted" warnings in browsers, why not :)

**Comment by justinmayer** [comment]
I see your point — I had forgotten about the "unencrypted" warnings that browsers added recently.

Re-direction is indeed [not yet supported](https://github.com/rtfd/readthedocs.org/issues/2652#issuecomment-413268346) but should automatically occur once RTD implements said re-direction in the future.
